core/src/core_tests/local_activities.rs

```diff
@@ -783,12 +776,19 @@ async fn test_schedule_to_start_timeout() {
     worker.run_until_done().await.unwrap();
 }
 
+#[rstest::rstest]
+#[case::sched_to_start(true)]
+#[case::sched_to_close(false)]
 #[tokio::test]
-async fn test_schedule_to_start_timeout_not_based_on_original_time() {
+async fn test_schedule_to_start_timeout_not_based_on_original_time(
```

Comment:
You should probably rename this test name to reflect both test cases. Maybe something along the line of `test_schedule_timeouts_...`